tests/test_cleaning.py

```diff
@@ -0,0 +1,234 @@
+                    transactionId="transaction_1",
+                    datetime=datetime(2022, 2, 1),
+                    amount=0.1,
+                    referenceText="reference_transaction_1",
+                    bankTransactionCode="dummy_transaction_code",
+                ),
+                SimpleTransaction(
+                    transactionId="transaction_2",
+                    datetime=datetime(2022, 2, 1),
+                    amount=0.3,
+                    referenceText="internal_reference_1",
+                    bankTransactionCode="dummy_transaction_code",
+                ),
+                SimpleTransaction(
+                    transactionId="transaction_3",
+                    datetime=datetime(2022, 2, 6),
+                    amount=-0.3,
+                    referenceText="internal_reference_1",
+                    bankTransactionCode="dummy_transaction_code",
+                ),
+            ],
+        ),
+        (
+            [
+                SimpleTransaction(
+                    transactionId="transaction_1",
+                    datetime=datetime(2022, 2, 1),
+                    amount=0.1,
+                    referenceText="reference_transaction_1",
+                    bankTransactionCode="dummy_transaction_code",
+                ),
+                SimpleTransaction(
+                    transactionId="transaction_2",
+                    datetime=datetime(2022, 2, 1),
+                    amount=0,
+                    referenceText="internal_reference_1",
+                    bankTransactionCode="dummy_transaction_code",
+                ),
+                SimpleTransaction(
+                    transactionId="transaction_3",
+                    datetime=datetime(2022, 2, 1),
+                    amount=-0.3,
+                    referenceText="internal_reference_1",
+                    bankTransactionCode="dummy_transaction_code",
+                ),
+            ],
+            [
+                SimpleTransaction(
+                    transactionId="transaction_1",
+                    datetime=datetime(2022, 2, 1),
+                    amount=0.1,
+                    referenceText="reference_transaction_1",
+                    bankTransactionCode="dummy_transaction_code",
+                ),
+                SimpleTransaction(
+                    transactionId="transaction_2",
+                    datetime=datetime(2022, 2, 1),
+                    amount=0,
+                    referenceText="internal_reference_1",
+                    bankTransactionCode="dummy_transaction_code",
+                ),
+                SimpleTransaction(
+                    transactionId="transaction_3",
+                    datetime=datetime(2022, 2, 1),
+                    amount=-0.3,
+                    referenceText="internal_reference_1",
+                    bankTransactionCode="dummy_transaction_code",
+                ),
+            ],
+        ),
+        (
+            [
```

Comment:
This looks like the code is doing the wrong thing, you might have caught a bug. **Three** transactions are wiped away from the original list, an internal transfer should remove pairs of transactions, which would be 2, 4, 6 etc.. transactions, never 3. 

That's a good catch through unit testing :)

Probably this line of code in `cleaning.py` needs a fix.
```python
        matching_transactions = [
            transaction
            for transaction in transactions   # <---- we need to remove `skipped_transactions` here as well, otherwise it matches the same transaction two times.
            if _has_internal_transfer_features(transaction)
            and current_amount == -transaction["amount"]
            and abs(current_datetime - transaction["datetime"])
            < timedelta(days=get_user_configuration().BankProcessingTimeInDays)
        ]

```